src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -507,22 +514,24 @@ void VM_PopulateDumpSharedSpace::doit() {
   MutexLocker ml(DumpTimeTable_lock, Mutex::_no_safepoint_check_flag);
   SystemDictionaryShared::check_excluded_classes();
 
-  StaticArchiveBuilder builder;
-  builder.gather_source_objs();
-  builder.reserve_buffer();
+  _builder.gather_source_objs();
+  if (_builder.reserve_buffer() == nullptr) {
+    this->_failed = true;
+    return;
```

Comment:
I think this should be removed, since reserve_buffer() will exit the JVM when OOM happens.

I did some experiments with `docket run --memory=1024m ...` and found that, at least on Linux, when a process runs over the memory limit, it just gets killed immediately by the OOM killer. There's no way to detect an OOM condition and handle it gracefully. So I think we can't (and don't need to) handle OOM errors.

Reply:
Isn't this just reservation? Can fail if you run out of address space (unlikely) or if you run against VMA limit.
Even for commit, you will fail if you run against commit limit or VMA number limit. In all these cases, no real memory has been used yet, so no reason to wake the OOM killer.